honeycomb-kernels/src/coloring/dsatur.rs

```diff
@@ -0,0 +1,95 @@
+use std::{cmp::Ordering, collections::HashMap};
+
+use honeycomb_core::{
+    cmap::{CMap2, DartIdType, Orbit2, OrbitPolicy, VertexIdType},
+    prelude::CoordsFloat,
+    stm::atomically,
+};
+use itertools::Itertools;
+use rayon::prelude::*;
+
+use super::Color;
+
+#[allow(clippy::missing_panics_doc)]
+/// DSATUR algorithm implementation
+///
+/// This algorithm is a coloring algorithm similar to the greedy coloring algorithm.
+///
+/// [Reference](https://en.wikipedia.org/wiki/DSatur).
+///
+/// # Return
+///
+/// The function will return the maximal used color, i.e. the `n_color_used - 1`.
+pub fn color<T: CoordsFloat>(cmap: &mut CMap2<T>) -> u8 {
```

Comment:
Not sure if it is a good name. Perhaps the algorithm name will be better. Add to the description it is a distance-1 coloring of the vertices.

Reply:
I re-export the function as `color_dsatur` in the `coloring` module. I can switch it to `dsatur` if you want